src/internal/SlidingSheet.js

```diff
@@ -0,0 +1,439 @@
+                 this.getMaxTranslatePrimary() -
+                  this.getTranslateMultiplier() * (this.swipeStartPrimary - currentPrimary),
+               0
+             ),
+             this.getMaxTranslatePrimary()
+           );
+  }
+
+  onBodyTouchMove = (event) => {
+    const currentX = event.touches[0].pageX;
+    const currentY = event.touches[0].pageY;
+    const vertical = this.isVertical();
+
+    const currentPrimary = vertical ? currentY : currentX;
+    const currentSecondary = vertical ? currentX : currentY;
+
+    if (this.state.swiping) {
+      event.preventDefault();
+      this.setPosition(this.getTranslatePrimary(currentPrimary));
+    } else if (this.maybeSwiping) {
+      const dPrimaryAbs = Math.abs(currentPrimary - this.touchStartPrimary);
+      const dSecondaryAbs = Math.abs(currentSecondary - this.touchStartSecondary);
+      // If the user has moved his thumb ten pixels in either direction,
+      // we can safely make an assumption about whether he was intending
+      // to swipe or scroll in the horizontal case
+      const threshold = 10;
+
+      if (dPrimaryAbs > threshold && dSecondaryAbs <= threshold) {
+        event.preventDefault();
+        this.swipeStartPrimary = currentPrimary;
+        this.setState({
+          swiping: this.props.open ? 'closing' : 'opening',
+        });
+        this.setPosition(this.getTranslatePrimary(currentPrimary));
+      } else if (dPrimaryAbs <= threshold && dSecondaryAbs > threshold) {
+        this.onBodyTouchEnd();
+      }
+    }
+  };
+
+  onBodyTouchEnd = (event) => {
+    if (this.state.swiping) {
+      const vertical = this.isVertical();
+      const currentPrimary = vertical ? event.changedTouches[0].pageY : event.changedTouches[0].pageX;
+      const translateRatio = this.getTranslatePrimary(currentPrimary) / this.getMaxTranslatePrimary();
+
+      this.maybeSwiping = false;
+      const swiping = this.state.swiping;
+      this.setState({
+        swiping: null,
+      });
+
+      // We have to open or close after setting swiping to null,
+      // because only then CSS transition is enabled.
+      if (translateRatio > 0.5) {
+        if (swiping === 'opening') {
+          this.setPosition(this.getMaxTranslatePrimary());
+        } else {
+          this.close('swipe');
+        }
+      } else {
+        if (swiping === 'opening') {
+          this.open('swipe');
+        } else {
+          this.setPosition(0);
+        }
+      }
+    } else {
+      this.maybeSwiping = false;
+    }
+
+    document.body.removeEventListener('touchmove', this.onBodyTouchMove);
+    document.body.removeEventListener('touchend', this.onBodyTouchEnd);
+    document.body.removeEventListener('touchcancel', this.onBodyTouchEnd);
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      containerClassName,
+      containerStyle,
+      direction,
+      modal,
+      overlayClassName,
+      overlayStyle,
+      style,
+      zDepth,
+    } = this.props;
```

Comment:
Same here. I would spread the other properties to the root component.
Who knows the need from of our users. Let's give them flexibility.